python_modules/dagster/dagster/_core/definitions/asset_in.py

```diff
@@ -77,3 +77,10 @@ def __new__(
                 else resolve_dagster_type(dagster_type)
             ),
         )
+
+    @classmethod
+    def from_coercible(cls, coercible: "CoercibleToAssetIn") -> "AssetIn":
+        return coercible if isinstance(coercible, AssetIn) else AssetIn(key=coercible)
```

Comment:
key = AssetKey.from_coercible, no?

Reply:
I think AssetIn is able to correctly coerce from the coercible